cmd/thv-operator/pkg/registryapi/config/config.go

```diff
@@ -42,20 +45,38 @@ func NewConfigManager(
 	if checksumManager == nil {
 		return nil, fmt.Errorf("checksumManager is required and cannot be nil")
 	}
-	return &configManager{client: k8sClient, scheme: scheme, checksum: checksumManager}, nil
+
+	return &configManager{client: k8sClient,
+		scheme:                      scheme,
+		checksum:                    checksumManager,
+		mcpRegistry:                 mcpRegistry,
+		registrySourceConfigMapName: "", // will be set when the config is built
```

Comment:
(we had this conversation with @dmartinol during the MVP development about checksum but I don't remember if we just decided to ignore the issue for now..)

It seems like storing state in structures that are to be used in an operator could lead to interesting races with multiple reconciles?

I think the way k8s operator reconcile loops work is that there is a guarantee that the same resource is never reconciled in parallel but there is a setting of maximum concurrent reconciles which, if bumped means that different resources can be reconciled at the same time which makes stateful structures problematic in k8s operators..